client/boot/common.js

```diff
@@ -427,13 +427,36 @@ const boot = ( currentUser, registerRoutes ) => {
 			registerRoutes();
 		}
 
+		const isDesktop = config.isEnabled( 'desktop' );
+		const loggedIn = currentUser.get() !== false;
+
+		const render = () => {
+			renderLayout( reduxStore );
+
+			page.start( { decodeURLComponents: false } );
+		};
+
 		// Render initial `<Layout>` for non-isomorphic sections.
 		// Isomorphic sections will take care of rendering their `<Layout>` themselves.
 		if ( ! document.getElementById( 'primary' ) ) {
-			renderLayout( reduxStore );
+			if ( ! isDesktop ) {
+				// If we're not in a WP-Desktop context, render.
+				render();
+			} else if ( loggedIn ) {
+				// WP-Desktop: logged in
+				// Wait on cookie-authentication
+				const ipc = require( 'electron' ).ipcRenderer;
+
+				ipc.on( 'cookie-auth-complete', function () {
+					debug( 'Desktop cookies set, rendering main layout...' );
+					render();
```

Comment:
Add a timeout to ensure we get some sort of render if this does not fire.

Reply:
Added timeout handling in e3a2ed791611fc1d5fd2a8eccdea9bb9423138b2.